starter_project/normalize_breton_test.py

```diff
@@ -1,15 +1,85 @@
+"""Tests for evaluating Celtic initial consonant mutations."""
+
 import unittest
 import normalize_breton_lib
 
 class TestStringMethods(unittest.TestCase):
+    """Tests for evaluating initial consonant mutations."""
 
     def test_normalize_breton(self):
-        'Test the output of NormalizeBreton.'
-        test_cases = [(('a--bc', 'a-bc'), ('ccb--a', 'ccb-a'), ('ba--aa', 'ba-aa'))]
-        for test in test_cases:
-          for test_case, expected in test:
-            test_fst = normalize_breton_lib.NormalizeBreton(test_case)
+        'Test the output of normalize_breton.'
+        with open("testdata/bre_normalized_sentences.tsv", "r") as test_file:
+            test_cases = test_file.readlines()[1:]
+#        test_cases = [(("ma gwin Da gwin tri pesk bara kozh ha KIG",
+#                        "ma gwin da win tri fesk bara kozh ha kig"),
+#                       ("da Kemper", "da gemper"))]
+        for sentence in test_cases:
+            test_case = sentence.strip().split("\t")[1]
+            expected = sentence.strip().split("\t")[2]
+            test_fst = normalize_breton_lib.normalize_breton(test_case)
             self.assertEqual(test_fst, expected)
 
+
+    def test_normalize_breton_soft_mutation(self):
+        'Test the Breton soft mutation.'
+        test_cases = [(("Div plac'h", "div blac'h"),
+                       ("DA TRA", "da dra"),
+                       ("da Kemper", "da gemper"),
+                       ("da gwin", "da win"),
+                       ("pe mamm", "pe vamm"))]
+        for test in test_cases:
+            for test_case, expected in test:
+                normalized_text = (test_case.strip().lower() @
+                                   normalize_breton_lib.DO_PREPROCESSING @
+                                   normalize_breton_lib.DO_SOFT_MUTATION @
+                                   normalize_breton_lib.DO_POSTPROCESSING
+                                   ).string()
```

Comment:
These are all the same apart from the relevant mutation type, make it a helper function.

Reply:
I added a helper function in `normalize_breton_lib.py` to do the pre/post-processing and for applying a single mutation. It actually looks like one I removed earlier, but it's probably better to keep it and keep the tests cleaner / more concise.